doc/developers/performance.rst

```diff
@@ -386,10 +386,17 @@ Checkout the official joblib documentation:
 - https://pythonhosted.org/joblib
 
 
-.. _warm-restarts:
+.. _warm-starts:
 
-A sample algorithmic trick: warm restarts for cross validation
+A sample algorithmic trick: warm starts for parameter search
 ==============================================================
 
-TODO: demonstrate the warm restart tricks for cross validation of linear
-regression with Coordinate Descent.
+One of the most common machine learning tasks is parameter search. 
+One approach is to sequentially train an estimator for a list of
+parameters on the same data. Since it is computationally
+expensive, one can speed up by using the `warm_start`
+parameter of the estimator. Setting `warm_start=True` allows to
+reuse the solution of the previous call to fit as initialization.
+In the case of linear model with Coordinate Descent, this technique
```

Comment:
This is the case not just in linear model with CD. The example of linear model with CD was probably here to just suggest that something like `ElasticNetCV` or `LogisticRegressionCV` etc does this for you.

But it would be good to include an example of code to perform warm restarts and evaluate performance on a held out set for each parameter, like a grid search.

Reply:
I added an example. It shows score and execution time when warm starting. 
For `warm_start=False` and `warm_start=True`, the best selected param remains the same but the score of warm starting is smaller.